app/views/courses/TeacherClassView.coffee

```diff
@@ -702,8 +702,15 @@ module.exports = class TeacherClassView extends RootView
       if status is 'enrolled' and student.prepaidType() is 'course'
         prepaid = student.makeCoursePrepaid()
         prepaid.revoke(student, {
-          success: =>
-            student.unset('coursePrepaid')
+          # The for loop completes before the success callback for the first student executes.
+          # So, the `student` will be the last student when the callback executes.
+          # Therefore, using a self calling anonymous function for the success callback
+          # to retain the student data for each iteration.
+          # Reference: https://www.pluralsight.com/guides/javascript-callbacks-variable-scope-problem
+          success: (() ->
+            st = student
+            return => st.unset('coursePrepaid')
```

Comment:
Is a `fat arrow` required?

Reply:
Also just for fun. You could pass the student into the anonymous function like so:

```
success: ((st) -> () -> st.unset('coursePrepaid'))(student)
```

Reply:
Which could then be simplified to:

```
success: (st -> -> st.unset('coursePrepaid'))(student)
```

Reply:
Yeah, interesting. :)
Not sure about the `fat arrow` question though. Do you think a `thin arrow` would also be fine?